### ISSUE TITLE
Unexpected dependencies

### ISSUE THREAD

**Issue by ThomWright** [eonarheim/TypeState#23]
Typestate has no dependencies listed in package.json, yet when I install it in a fresh project with no other dependencies, I get this:

```
typestate-test@1.0.0 /path/to/project
└─┬ typestate@1.0.5
  └─┬ grunt-jasmine-node@0.3.1
    ├── coffee-script@1.6.3
    └─┬ jasmine-node@1.14.5
      ├── coffee-script@1.6.3 deduped
      ├─┬ gaze@0.3.4
      │ ├─┬ fileset@0.1.8
      │ │ ├─┬ glob@3.2.11
      │ │ │ ├── inherits@2.0.3
      │ │ │ └─┬ minimatch@0.3.0
      │ │ │   ├── lru-cache@2.7.3 deduped
      │ │ │   └── sigmund@1.0.1 deduped
      │ │ └── minimatch@0.2.14 deduped
      │ └─┬ minimatch@0.2.14
      │   ├── lru-cache@2.7.3
      │   └── sigmund@1.0.1
      ├─┬ jasmine-growl-reporter@0.0.3
      │ └── growl@1.7.0
      ├─┬ jasmine-reporters@1.0.2
      │ └── mkdirp@0.3.5 deduped
      ├── mkdirp@0.3.5
      ├── requirejs@2.3.5
      ├── underscore@1.8.3
      └── walkdir@0.0.12
```

Any idea why?

**Comment by kamranayub** [comment]
This is just how npm works, I suppose. From the docs:

"By default, npm install will install all modules listed as dependencies in
package.json.

With the --production flag (or when the NODE_ENV environment variable is
set to production), npm will not install modules listed in devDependencies."

https://docs.npmjs.com/cli/install

It won't have any real impact on your development but it looks like you can
pass a flag and disable that.

**Comment by ThomWright** [comment]
Maybe this is a weird NPM thing, but not one I've seen before.

- `typestate` is my only dependency
- `grunt-jasmine-node` is the only dependency installed for `typestate` (none of its devDependencies are installed)
- `grunt-jasmine-node` is not listed anywhere in `typestate`'s package.json
- `grunt-jasmine-node` gets installed even if I run `npm install --production`

Where is it coming from? :confused:

**Comment by kamranayub** [comment]
I have no idea. This is what I get when I clone fresh and do an npm install, there's no `grunt-jasmine-node`.

```
typestate@1.0.5 /Users/z002s6f/dev/contrib/TypeState
├─┬ grunt@0.4.2
│ ├── async@0.1.22
│ ├── coffee-script@1.3.3
│ ├── colors@0.6.2
│ ├── dateformat@1.0.2-1.2.3
│ ├── eventemitter2@0.4.14
│ ├── exit@0.1.2
│ ├─┬ findup-sync@0.1.3
│ │ ├─┬ glob@3.2.11
│ │ │ ├── inherits@2.0.3 deduped
│ │ │ └─┬ minimatch@0.3.0
│ │ │   ├── lru-cache@2.7.3 deduped
│ │ │   └── sigmund@1.0.1 deduped
│ │ └── lodash@2.4.2
│ ├── getobject@0.1.0
│ ├─┬ glob@3.1.21
│ │ ├── graceful-fs@1.2.3
│ │ ├── inherits@1.0.2
│ │ └── minimatch@0.2.14 deduped
│ ├── hooker@0.2.3
│ ├── iconv-lite@0.2.11
│ ├─┬ js-yaml@2.0.5
│ │ ├─┬ argparse@0.1.16
│ │ │ ├── underscore@1.7.0
│ │ │ └── underscore.string@2.4.0
│ │ └── esprima@1.0.4
│ ├── lodash@0.9.2
│ ├─┬ minimatch@0.2.14
│ │ ├── lru-cache@2.7.3
│ │ └── sigmund@1.0.1
│ ├─┬ nopt@1.0.10
│ │ └── abbrev@1.1.1
│ ├─┬ rimraf@2.0.3
│ │ └── graceful-fs@1.1.14
│ ├── underscore.string@2.2.1
│ └── which@1.0.9
├── grunt-contrib-concat@0.1.2
├── grunt-contrib-copy@0.5.0
├─┬ grunt-contrib-watch@0.6.1
│ ├── async@0.2.10
│ ├─┬ gaze@0.5.2
│ │ └─┬ globule@0.1.0
│ │   ├── glob@3.1.21 deduped
│ │   ├── lodash@1.0.2
│ │   └── minimatch@0.2.14 deduped
│ ├── lodash@2.4.2
│ └─┬ tiny-lr-fork@0.0.5
│   ├── debug@0.7.4
│   ├── faye-websocket@0.4.4
│   ├─┬ noptify@0.0.3
│   │ └─┬ nopt@2.0.0
│   │   └── abbrev@1.1.1 deduped
│   └── qs@0.5.6
├─┬ grunt-minified@0.0.6
│ ├── colors@0.6.2 deduped
│ └─┬ uglify-js@2.2.5
│   ├─┬ optimist@0.3.7
│   │ └── wordwrap@0.0.3
│   └─┬ source-map@0.1.43
│     └── amdefine@1.0.1
├─┬ grunt-shell@0.6.1
│ └─┬ chalk@0.3.0
│   ├── ansi-styles@0.2.0
│   └── has-color@0.1.7
├─┬ jasmine@2.99.0
│ ├── exit@0.1.2 deduped
│ ├─┬ glob@7.1.2
│ │ ├── fs.realpath@1.0.0
│ │ ├─┬ inflight@1.0.6
│ │ │ ├── once@1.4.0 deduped
│ │ │ └── wrappy@1.0.2
│ │ ├── inherits@2.0.3
│ │ ├─┬ minimatch@3.0.4
│ │ │ └─┬ brace-expansion@1.1.11
│ │ │   ├── balanced-match@1.0.0
│ │ │   └── concat-map@0.0.1
│ │ ├─┬ once@1.4.0
│ │ │ └── wrappy@1.0.2 deduped
│ │ └── path-is-absolute@1.0.1
│ └── jasmine-core@2.99.1
└── typescript@1.8.10
```

**Comment by ThomWright** [comment]
That's the strange thing. My steps were (using npm 5.7.1):

- `mkdir typestate-test && cd typestate-test`
- `npm init` and accept defaults
- `npm install --save typestate`
- `npm ls`

It's about as clean an environment as I can make.

If I look in `node_modules/typestate/package.json` I see:

```
  "dependencies": {
    "grunt-jasmine-node": "^0.3.1"
  },
```

So the published package seems to have it as a dependency.

Also, unrelated, but the published package includes lots of file it probably shouldn't:

- a 1.6MB binary: `tools/NuGet.exe`
- `Gruntfile.js`
- `src/`
- `spec/`
- `example/`

**Comment by kamranayub** [comment]
OKAY! I know what's up. @eonarheim it looks like the package.json on npm is out-of-date from master!

https://www.npmjs.com/package/typestate

See it lists `grunt-jasmine-node`? That's the issue. It just needs to reflect latest version.

**Comment by ThomWright** [comment]
Cool! Thanks for sticking with it :slightly_smiling_face:

**Comment by eonarheim** [comment]
@ThomWright Just published a new version of typestate@1.0.6 let me know if you run into issues
